source/store/test/AppCompatibilityContext.test.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  AppCompatibilityVisitor,
+  APPLICATION_COMPATIBILITY_STATUS,
+} from "app/types/AppCompatibilityTypes";
+import { createAppCompatibilityContextValue } from "../AppCompatibilityContext";
+
+describe("AppCompatibilityContext", () => {
+  it("visit(visitor) calls correct visitor method", () => {
+    const visit = (
+      status: APPLICATION_COMPATIBILITY_STATUS,
+      updateUrl: string,
+      visitor: Partial<AppCompatibilityVisitor<unknown>>
+    ) => {
+      let c = createAppCompatibilityContextValue({
+        status: status,
+        updateUrl: updateUrl,
+      });
+      return c.visit(visitor);
+    };
+
+    expect(
+      visit(APPLICATION_COMPATIBILITY_STATUS.COMPATIBLE, "", {
+        compatible: () => "compatible was called",
+      })
```

Comment:
I would prefer to extract the value to be expected on to a separate variable, so that the expect statement reads more clearly 🙂. I also think it would be better/clearer to just `jest.fn()` and `.toHaveBeenCalled()` since the actual value is not important.

```typescript
const compatible = jest.fn()

visit(..., { compatible })

expect(compatible).toHaveBeenCalled()
```

If we care about the return value I think you can modify and expect on the `jest.fn()` return value (not too sure about this; I haven't used `jest.fn()` that much).

(same goes for the other `expect` calls).